src/test.js

```diff
@@ -737,23 +741,41 @@ function internalStart( test ) {
 	}
 }
 
-function numberOfTests( module ) {
-	let count = module.tests.length;
+function collectTests( module, key ) {
+	const tests = [].concat( module[ key ] );
 	const modules = [ ...module.childModules ];
 
 	// Do a breadth-first traversal of the child modules
 	while ( modules.length ) {
 		const nextModule =  modules.shift();
-		count += nextModule.tests.length;
+		tests.push.apply( tests, nextModule[ key ] );
```

Comment:
I'd prefer to keep this as a counter instead of an array as we don't need the flattened array for anything other than length.